builder/vsphere/driver/vm_mock.go

```diff
@@ -182,10 +182,22 @@ func (vm *VirtualMachineMock) GetDir() (string, error) {
 	return vm.GetDirResponse, vm.GetDirErr
 }
 
-func (vm *VirtualMachineMock) AddCdrom(cdromType string, isoPath string) error {
+func (vm *VirtualMachineMock) GetOrMakeCdroms(reuseCDrom bool, controllerType string, n_cdroms int) (object.VirtualDeviceList, error) {
+	var arr_cdroms object.VirtualDeviceList
```

Comment:
Okay, I have a question which I think it might be faster to ask here, especially because I'm not sure my solution with interfaces might be correct.

The code currently won't compile because I need to adjust tests, which is where I'm stuck.

I have added a `GetOrMakeCdroms` function to `VirtualMachine interface`, which returns a `(VirtualDeviceList, error)`. Now, in `vm_mock.go` file we re-implement the interface for testing purposes. So for example a mock `AddCdrom` function *(which I renamed to `MountCdrom`)* adds a string to some internal variable. In case of the new `GetOrMakeCdroms` I can't just add a string but I also need to create a `VirtualDeviceList`, which is an alias to a `BaseVirtualDevice interface`. So what's the proper course of action here, should I reimplement the `BaseVirtualDevice` as well?

Reply:
I just tried to work around that by declaring a `type VirtualDeviceList []string`, but apparently Go wouldn't consider such function an implementation of the `interface`. It wants it to be a `object.VirtualDeviceList`, not some local type `VirtualDeviceList`.

Reply:
Okay, I solved that locally by creating a wrapper `VirtualDeviceMock` and implementing the `interface` *(which thankfully only had a single function, so that didn't take too long)*. Marking as solved